### PR TITLE
gate: remove method

### PR DESCRIPTION
**Pull request by sc0Vu**

### CODE REVIEW

File: ts/src/gate.ts [hunk 1]
```diff
@@ -5028,7 +5112,14 @@ export default class gate extends Exchange {
         } else {
             request['leverage'] = leverage.toString ();
         }
-        const response = await this[method] (this.extend (request, query));
+        let response = undefined;
+        if (market['type'] === 'swap') {
```
Comment (carlosmiei) [inline ts/src/gate.ts:5116]: we can use `market['swap']` here directly

File: ts/src/gate.ts [hunk 2]
```diff
@@ -5028,7 +5112,14 @@ export default class gate extends Exchange {
...
+            response = await this.privateFuturesPostSettlePositionsContractLeverage (this.extend (request, query));
+        } else if (market['type'] === 'future') {
```
Comment (carlosmiei) [inline ts/src/gate.ts:5118]: and `market['future']`

File: ts/src/gate.ts [hunk 3]
```diff
@@ -5880,11 +5974,14 @@ export default class gate extends Exchange {
         const market = this.market (symbol);
         const [ request, query ] = this.prepareRequest (market, undefined, params);
         request['change'] = this.numberToString (amount);
-        const method = this.getSupportedMapping (market['type'], {
-            'swap': 'privateFuturesPostSettlePositionsContractMargin',
-            'future': 'privateDeliveryPostSettlePositionsContractMargin',
-        });
-        const response = await this[method] (this.extend (request, query));
+        let response = undefined;
+        if (market['type'] === 'swap') {
```
Comment (carlosmiei) [inline ts/src/gate.ts:5978]: same here

File: ts/src/gate.ts (hunk 1 above)
Comment (sc0Vu) [reply ts/src/gate.ts:5116]: @carlosmiei how about update all type checking for swap, future, option to `market[key]`?

**Comment by carlosmiei** [comment]
DEMO

```
p gate createOrder "LTC/USDT" limit buy 0.1 50         
Python v3.11.5
CCXT v4.1.67
gate.createOrder(LTC/USDT,limit,buy,0.1,50)
{'amount': 0.1,
 'average': None,
 'clientOrderId': 'apiv4',
 'cost': 0.0,
 'datetime': '2023-11-29T16:26:36.828Z',
 'fee': None,
 'fees': [],
 'filled': 0.0,
 'id': '447200283432',
 'info': {'account': 'spot',
          'amend_text': '-',
          'amount': '0.1',
          'biz_info': 'ch:ccxt',
          'create_time': '1701275196',
          'create_time_ms': '1701275196828',
          'currency_pair': 'LTC_USDT',
          'fee': '0',
          'fee_currency': 'LTC',
          'fill_price': '0',
          'filled_total': '0',
          'finish_as': 'open',
          'gt_discount': False,
          'gt_fee': '0',
          'gt_maker_fee': '0',
          'gt_taker_fee': '0',
          'iceberg': '0',
          'id': '447200283432',
          'left': '0.1',
          'point_fee': '0',
          'price': '50',
          'rebated_fee': '0',
          'rebated_fee_currency': 'USDT',
          'side': 'buy',
          'status': 'open',
          'text': 'apiv4',
          'time_in_force': 'gtc',
          'type': 'limit',
          'update_time': '1701275196',
          'update_time_ms': '1701275196828'},
 'lastTradeTimestamp': 1701275196828,
 'lastUpdateTimestamp': None,
 'postOnly': False,
 'price': 50.0,
 'reduceOnly': None,
 'remaining': 0.1,
 'side': 'buy',
 'status': 'open',
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': 'GTC',
 'timestamp': 1701275196828,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```
```
p gate fetchMyTrades "LTC/USDT" None 1  
Python v3.11.5
CCXT v4.1.67
gate.fetchMyTrades(LTC/USDT,None,1)
[{'amount': 0.0864,
  'cost': 5.998752,
  'datetime': '2023-11-29T16:27:34.132Z',
  'fee': None,
  'fees': [{'cost': '0.0001728', 'currency': 'LTC'},
           {'cost': '0.0', 'currency': 'GT'},
           {'cost': '0.0', 'currency': 'GatePoint'}],
  'id': '6767737261',
  'info': {'amend_text': '-',
           'amount': '0.0864',
           'create_time': '1701275254',
           'create_time_ms': '1701275254132.880000',
           'currency_pair': 'LTC_USDT',
           'fee': '0.0001728',
           'fee_currency': 'LTC',
           'gt_fee': '0.0',
           'id': '6767737261',
           'order_id': '447200789457',
           'point_fee': '0.0',
           'price': '69.43',
           'role': 'taker',
           'sequence_id': '713276',
           'side': 'buy',
           'text': 'apiv4'},
  'order': '447200789457',
  'price': 69.43,
  'side': 'buy',
  'symbol': 'LTC/USDT',
  'takerOrMaker': 'taker',
  'timestamp': 1701275254132,
  'type': None}]
```
```
 p gate fetchOrder "447200789457" "LTC/USDT"                 
Python v3.11.5
CCXT v4.1.67
gate.fetchOrder(447200789457,LTC/USDT)
{'amount': 0.08641797493878726,
 'average': 69.43,
 'clientOrderId': 'apiv4',
 'cost': 6.0,
 'datetime': '2023-11-29T16:27:34.136Z',
 'fee': {'cost': 0.0001728, 'currency': 'LTC'},
 'fees': [{'cost': 0.0001728, 'currency': 'LTC'}],
 'filled': 0.0864,
 'id': '447200789457',
 'info': {'account': 'spot',
          'amend_text': '-',
          'amount': '6',
          'avg_deal_price': '69.43',
          'biz_info': 'ch:ccxt',
          'create_time': '1701275254',
          'create_time_ms': '1701275254136',
          'currency_pair': 'LTC_USDT',
          'fee': '0.0001728',
          'fee_currency': 'LTC',
          'fill_price': '5.998752',
          'filled_total': '5.998752',
          'finish_as': 'filled',
          'gt_discount': False,
          'gt_fee': '0',
          'gt_maker_fee': '0',
          'gt_taker_fee': '0',
          'iceberg': '0',
          'id': '447200789457',
          'left': '0.001248',
          'point_fee': '0',
          'price': '0',
          'rebated_fee': '0',
          'rebated_fee_currency': 'USDT',
          'side': 'buy',
          'status': 'closed',
          'text': 'apiv4',
          'time_in_force': 'ioc',
          'type': 'market',
          'update_time': '1701275254',
          'update_time_ms': '1701275254136'},
 'lastTradeTimestamp': 1701275254136,
 'lastUpdateTimestamp': None,
 'postOnly': False,
 'price': 69.43,
 'reduceOnly': None,
 'remaining': 1.7974938787267e-05,
 'side': 'buy',
 'status': 'closed',
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': 'IOC',
 'timestamp': 1701275254136,
 'trades': [],
 'triggerPrice': None,
 'type': 'market'}
```
```
 p gate cancelOrder "447200283432" "LTC/USDT"            
Python v3.11.5
CCXT v4.1.67
gate.cancelOrder(447200283432,LTC/USDT)
{'amount': 0.1,
 'average': None,
 'clientOrderId': 'apiv4',
 'cost': 0.0,
 'datetime': '2023-11-29T16:26:36.828Z',
 'fee': None,
 'fees': [],
 'filled': 0.0,
 'id': '447200283432',
 'info': {'account': 'spot',
          'amend_text': '-',
          'amount': '0.1',
          'biz_info': 'ch:ccxt',
          'create_time': '1701275196',
          'create_time_ms': '1701275196828',
          'currency_pair': 'LTC_USDT',
          'fee': '0',
          'fee_currency': 'LTC',
          'fill_price': '0',
          'filled_total': '0',
          'finish_as': 'cancelled',
          'gt_discount': False,
          'gt_fee': '0',
          'gt_maker_fee': '0',
          'gt_taker_fee': '0',
          'iceberg': '0',
          'id': '447200283432',
          'left': '0.1',
          'point_fee': '0',
          'price': '50',
          'rebated_fee': '0',
          'rebated_fee_currency': 'USDT',
          'side': 'buy',
          'status': 'cancelled',
          'text': 'apiv4',
          'time_in_force': 'gtc',
          'type': 'limit',
          'update_time': '1701275344',
          'update_time_ms': '1701275344272'},
 'lastTradeTimestamp': 1701275344272,
 'lastUpdateTimestamp': None,
 'postOnly': False,
 'price': 50.0,
 'reduceOnly': None,
 'remaining': 0.1,
 'side': 'buy',
 'status': 'canceled',
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': 'GTC',
 'timestamp': 1701275196828,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```
```
 p gate fetchBalance                         
Python v3.11.5
CCXT v4.1.67
gate.fetchBalance()
{'ADA': {'free': 9.639862355644, 'total': 9.639862355644, 'used': 0.0},
 'BTC': {'free': 9.98e-05, 'total': 9.98e-05, 'used': 0.0},
 'DOGE': {'free': 10.1, 'total': 10.1, 'used': 0.0},
 'LTC': {'free': 0.857570579705, 'total': 0.857570579705, 'used': 0.0},
 'TON': {'free': 1.63778073, 'total': 1.63778073, 'used': 0.0},
 'USDT': {'free': 35.993110081036, 'total': 35.993110081036, 'used': 0.0},
 'XRP': {'free': 13.470006, 'total': 13.470006, 'used': 0.0},
 'free': {'ADA': 9.639862355644,
```

**Pull request merged**